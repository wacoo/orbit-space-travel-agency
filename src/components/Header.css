.header {
  width: 100%;
  height: 70px;
  display: flex;
  justify-content: space-between;
  background-color: lightgray;
}

.logo-title {
  display: flex;
  align-items: center;
}

.header img {
  width: 50px;
  margin: 0 5px 0 24px;
}

.header h1 {
  margin: 0;
}

.header ul {
  width: 20%;
  list-style-type: none;
  display: flex;
  justify-content: center;
  align-items: center;
}

.header li {
  margin-right: 24px;
  padding: 8px 12px;
}

.header a {
  text-decoration: none;
  color: rgb(80, 90, 238);
}
