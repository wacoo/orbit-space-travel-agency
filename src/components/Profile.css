.profile-container {
  width: 100%;
  margin-top: 10vh;
  display: flex;
  justify-content: center;
  align-items: center;
}

.profile-wrapper {
  display: flex;
  width: 90%;
  padding: 24px;
  background-color: aliceblue;
  border-radius: 10px;
}

.my-rockets,
.my-mission {
  width: 100%;
  border-collapse: collapse;
}

.my-rockets tr {
  margin: 0 24px;
}

.my-rockets th {
  height: 50px;
}

.my-rockets td {
  width: 45%;
  height: 100px;
  text-align: center;
  border: gray solid 2px;
}
