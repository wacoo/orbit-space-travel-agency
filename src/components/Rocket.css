.card {
  width: 100%;
  min-height: 100px;
  display: flex;
}

.card-image,
.card-text {
  margin: 12px;
}

.card img {
  width: 200px;
  height: 150px;
  background-color: aqua;
}

.card-text {
  display: flex;
  flex-direction: column;
}

.card button {
  margin: 12px;
  margin-left: 0;
}
