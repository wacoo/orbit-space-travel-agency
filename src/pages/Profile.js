import React from 'react';

const MyProfileContainer = () => (
  <div className="profileContainer" />
);

export default MyProfileContainer;
