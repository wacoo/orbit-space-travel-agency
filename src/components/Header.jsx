import './Header.css';
import Logo from '../img/logo.png';

const Header = () => (
  <header className="header" >
    <div className="logo-title">
      <img src={Logo} alt="Logo" />
      <h1>rbit Space Travelers&apos; Hub</h1>
    </div>
  </header>
);

export default Header;
