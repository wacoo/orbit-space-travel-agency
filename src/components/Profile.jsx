import { useSelector } from 'react-redux';
import './Profile.css';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets.rockets);
  return (
    <div className="profile-container">
      <div className="profile-wrapper">
        <table className="my-mission">
          <thead>
            <tr>
              <th>My Missions</th>
            </tr>
          </thead>
          {/* <tbody>
            {missions.map((mission) => (
              mission.reserved ? (
                <tr key={mission.id ? mission.id : null}>
                  <td>
                    <p key={mission.id}>{mission.rocket_name}</p>
                  </td>
                </tr>
              )
                : null
            ))}
          </tbody> */}
        </table>
        <table className="my-rockets">
          <thead>
            <tr>
              <th>My Rockets</th>
            </tr>
          </thead>
          <tbody>
            {rockets.map((rocket) => (
              rocket.reserved ? (
                <tr key={rocket.id ? rocket.id : null}>
                  <td>
                    <p key={rocket.id}>{rocket.rocket_name}</p>
                  </td>
                </tr>
              )
                : null
            ))}
          </tbody>
        </table>
        <div className="my-missions" />
      </div>
    </div>
  );
};

export default Profile;
