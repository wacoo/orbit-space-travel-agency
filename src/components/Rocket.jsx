import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import './Rocket.css';
import { cancelRocketReservation, fetchRockets, reserveRocket } from '../features/rockets/rocketsSlice';

const Rocket = () => {
  const dispatch = useDispatch();
  const rockets = useSelector((state) => state.rockets.rockets);

  useEffect(() => {
    dispatch(fetchRockets());
  }, [dispatch]);

  const handleReserveRocket = (id) => {
    dispatch(reserveRocket(id));
  };

  const cancelReservation = (id) => {
    dispatch(cancelRocketReservation(id));
  };

  return rockets.map((rocket) => (
    <div className="card" key={rocket.id}>
      <div className="card-image">
        <img src={rocket.flickr_images[1]} alt="" />
      </div>
      <div className="card-text">
        <h2 className="rocket-name">{rocket.rocket_name}</h2>
        <p className="rocket-desc">
          {rocket.description}
        </p>
        <div>
          {rocket.reserved ? (<button type="button" onClick={() => cancelReservation(rocket.id)}>Cancel Reservation</button>)
            : (<button type="button" onClick={() => handleReserveRocket(rocket.id)}>Reserve Rocket</button>)}
        </div>
      </div>
    </div>
  ));
};

export default Rocket;
