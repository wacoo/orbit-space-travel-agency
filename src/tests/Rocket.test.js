import React from 'react';
import renderer from 'react-test-renderer';
import { useDispatch, useSelector } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import Header from '../components/Header';
import Rocket from '../components/Rocket';
import Profile from '../components/Profile';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}));
let res = '';
const handleClick = () => {
  res = 'Clicked!';
};

const stateArray = [
  {
    id: '1234',
    rocket_name: 'Falcon 1',
    description: 'The first to Mars',
    flickr_images: '',
  },
  {
    id: '5667',
    rocket_name: 'Star Ship',
    description: 'The first to Mars',
    flickr_images: '',
  },
  {
    id: '2345',
    rocket_name: 'Falcon Heavy',
    description: 'The first to the moon',
    flickr_images: '',
    reserved: true,
  },
  {
    id: '2349',
    rocket_name: 'Falcon Light',
    description: 'The first to the moon',
    flickr_images: '',
    reserved: true,
  },
];

describe('Test rockets component', () => {
  it('Tests Header', () => {
    const headerComponent = renderer.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(headerComponent).toMatchSnapshot();
  });

  it('Tests Rockets', () => {
    useDispatch();
    useSelector.mockReturnValue(stateArray);
    const component = renderer.create(
      <BrowserRouter>
        <Rocket />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Tests Profile', () => {
    useDispatch();
    useSelector.mockReturnValue(stateArray);
    const component = renderer.create(
      <BrowserRouter>
        <Profile />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Test user interaction', () => {
  it('Tests Reserve Rocket Button', () => {
    useSelector.mockReturnValue(stateArray);
    render(<BrowserRouter><Rocket /></BrowserRouter>);
    const reserveButton = screen.getAllByText('Reserve Rocket');
    reserveButton[0].onclick = handleClick();
    fireEvent.click(reserveButton[0]);
    expect(res).toBe('Clicked!');
  });

  it('Tests Cancel Reservation Button', () => {
    useSelector.mockReturnValue(stateArray);
    render(<BrowserRouter><Rocket /></BrowserRouter>);
    const reserveButton = screen.getAllByText('Cancel Reservation');
    reserveButton[0].onclick = handleClick();
    fireEvent.click(reserveButton[0]);
    expect(res).toBe('Clicked!');
  });
});
