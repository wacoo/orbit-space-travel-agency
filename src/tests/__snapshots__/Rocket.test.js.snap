// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rockets component Tests Header 1`] = `
<nav
  className="header-nav"
>
  <div
    className="logo-container"
  >
    <img
      alt="logo"
      className="site-logo"
      src="logo.png"
    />
    <h1>
      Space Travelers' Hub
    </h1>
  </div>
  <ul
    className="nav-items"
  >
    <li>
      <a
        aria-current="page"
        className="nav-link active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
    </li>
    <li>
      <a
        className="icon nav-link"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
    </li>
    <li>
      <a
        className="nav-link"
        href="/profile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Test rockets component Tests Profile 1`] = `
<div
  className="profile-container"
>
  <div
    className="profile-wrapper"
  >
    <table
      className="my-rockets"
    >
      <thead>
        <tr>
          <th>
            My Missions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
    <table
      className="my-rockets"
    >
      <thead>
        <tr>
          <th>
            My Rockets
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <p>
              Falcon Heavy
            </p>
          </td>
        </tr>
        <tr>
          <td>
            <p>
              Falcon Light
            </p>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Test rockets component Tests Rockets 1`] = `
Array [
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt=""
      />
    </div>
    <div
      className="card-text"
    >
      <h2>
        Falcon 1
      </h2>
      <p
        className="rocket-desc"
      >
        The first to Mars
      </p>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>,
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt=""
      />
    </div>
    <div
      className="card-text"
    >
      <h2>
        Star Ship
      </h2>
      <p
        className="rocket-desc"
      >
        The first to Mars
      </p>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          Reserve Rocket
        </button>
      </div>
    </div>
  </div>,
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt=""
      />
    </div>
    <div
      className="card-text"
    >
      <h2>
        Falcon Heavy
      </h2>
      <p
        className="rocket-desc"
      >
        <span
          className="reserved"
        >
          Reserved
        </span>
        The first to the moon
      </p>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </div>
  </div>,
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <img
        alt=""
      />
    </div>
    <div
      className="card-text"
    >
      <h2>
        Falcon Light
      </h2>
      <p
        className="rocket-desc"
      >
        <span
          className="reserved"
        >
          Reserved
        </span>
        The first to the moon
      </p>
      <div>
        <button
          onClick={[Function]}
          type="button"
        >
          Cancel Reservation
        </button>
      </div>
    </div>
  </div>,
]
`;
